### PR TITLE
feat: added star-fusion

### PR DESCRIPTION
**Pull request by jonca79**

### CODE REVIEW

File: workflow/rules/star_fusion.smk [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        fq1="prealignment/merged/{sample}_{type}_fastq1.fastq.gz",
+        fq2="prealignment/merged/{sample}_{type}_fastq2.fastq.gz",
+        genome_path=config.get("star_fusion", {}).get("genome_path", ""),
+    output:
+        fusions=temp("fusions/star_fusion/{sample}_{type}/star-fusion.fusion_predictions.tsv"),
+        fusions_abridged=temp("fusions/star_fusion/{sample}_{type}/star-fusion.fusion_predictions.abridged.tsv"),
+        bam=temp("fusions/star_fusion/{sample}_{type}/Aligned.out.bam"),
+        sj=temp("fusions/star_fusion/{sample}_{type}/SJ.out.tab"),
+    params:
+        output_dir=temp(directory("fusions/star_fusion/{sample}_{type}/")),
```
Comment (marrip) [inline workflow/rules/star_fusion.smk:21]: alphabetical order

File: .tests/integration/config.yaml [hunk 2]
```diff
@@ -2,11 +2,16 @@ resources: "resources.yaml"
...
 reference:
   fasta: "reference/HES45.fna"
+  star_fusion: "reference/HES45.fna"
 
 gene_fuse:
   container: "docker://hydragenetics/genefuse:0.6.1"
   genes: "reference/gene_fuse_HES4.hg19.csv"
+
+star_fusion:
+  container: "docker://hydragenetics/star-fusion:1.10.1"
```
[the remark this answers is not in the document]
Comment (jonca79) [reply .tests/integration/config.yaml:16]: Yes, it works!

[the remark this answers is not in the document]
Comment (marrip) [reply .tests/integration/config.yaml:16]: WHOOOT?! 🎉

**Review by Smeds — APPROVED** [review]

File: workflow/schemas/config.schema.yaml [hunk 3]
```diff
@@ -18,30 +18,35 @@ properties:
     - resources
     - default_container
 
+
   gene_fuse:
     type: object
```
Comment (marrip) [inline workflow/schemas/config.schema.yaml:23]: description is missing

File: workflow/schemas/config.schema.yaml [hunk 4]
```diff
@@ -18,30 +18,35 @@ properties:
...
       extra:
         type: string
-        description: extra parameters to gene_fuse
+        description: extra parameters to rule
       required:
         - genes
+        - fasta
 
     gene_fuse_report:
       type: object
```
Comment (marrip) [inline workflow/schemas/config.schema.yaml:45]: description is missing

File: workflow/schemas/config.schema.yaml [hunk 5]
```diff
@@ -51,5 +56,21 @@ properties:
         unique_read_limit:
           type: integer
           description: Lower limit (not inclusive) of unique reads needed for fusion to be included in report
-        required:
-          - genes
+
+  star_fusion:
+    type: object
```
Comment (marrip) [inline workflow/schemas/config.schema.yaml:61]: description is missing

File: workflow/schemas/resources.schema.yaml [hunk 6]
```diff
@@ -20,43 +15,71 @@ properties:
...
     - default_resources
 
   gene_fuse:
     type: object
+    description: resource definitions for gene_fuse
     properties:
       threads:
         type: integer
-        description: number of threads that will be used by 'gene_fuse'
+        description: number of threads that will be used by rule
```
Comment (marrip) [inline workflow/schemas/resources.schema.yaml:33]: I think you can skip "for rule" and "by rule". It is somehow implied that it is connected to a specific rule by the hierarchical order.

File: workflow/schemas/resources.schema.yaml [hunk 7]
```diff
@@ -20,43 +15,71 @@ properties:
...
-        description: memory used for 'gene_fuse_report'
+        description: memory used for rule
       mem_per_cpu:
         type: integer
-        description: memory used per cpu for 'gene_fuse_report'
+        description: memory used per cpu for rule
       partition:
         type: string
-        description: partition to use on the cluser for 'gene_fuse_report'
+        description: partition to use on the cluser for rule
```
Comment (marrip) [inline workflow/schemas/resources.schema.yaml:79]: cluster

File: workflow/schemas/resources.schema.yaml [hunk 8]
```diff
@@ -20,43 +15,71 @@ properties:
...
     properties:
+      mem_mb:
+        type: integer
+        description: memory used for rule
+      mem_per_cpu:
+        type: integer
+        description: memory used per cpu for rule
+      partition:
+        type: string
+        description: partition to use on the cluser for rule
```
Comment (marrip) [inline workflow/schemas/resources.schema.yaml:59]: cluster

File: workflow/rules/star_fusion.smk [hunk 9]
```diff
@@ -0,0 +1,51 @@
+# vim: syntax=python tabstop=4 expandtab
```
Comment (marrip) [inline workflow/rules/star_fusion.smk:1]: I still think these syntaxing comments are misplaced as they are very specific for one user and should not be part of the code base that is shared among different users.

File: workflow/schemas/resources.schema.yaml [hunk 10]
```diff
@@ -14,7 +14,7 @@ properties:
         description: default memory used per cpu for a rule
       partition:
         type: string
-        description: default partition to use on the cluser for a rule
+        description: default partition to use on the clusetr for a rule
```
Comment (marrip) [inline workflow/schemas/resources.schema.yaml:17]: almost 😜

**Review by marrip — APPROVED** [review]

**Comment by marrip** [comment]
Just a general comment, the integration test will fail as there is no reference for `star_fusion`, right? I guess, we can leave it for now and look at it later?

**Comment by jonca79** [comment]
Yes, we should leave that for later.

**Pull request merged**